### PR TITLE
Services:WebDAV - GUI to Middlewared

### PR DESCRIPTION
**Pull request by skarekrow**
Ticket: #25943

### CODE REVIEW

File: gui/services/forms.py [hunk 1]
```diff
@@ -1808,32 +1783,17 @@ def clean(self):
...
-        return obj
+    def middleware_clean(self, data):
+        data['protocol'] = data['protocol'].upper()
+        data['htauth'] = data['htauth'].upper()
+
+        return data
 
     def done(self, *args, **kwargs):
-        if self._has_changed('webdav_certssl'):
+        if getattr(self.instance, 'webdav_certssl') != 'NONE':
```
Comment (themylogin) [inline gui/services/forms.py]: This should happend in middleware too

File: gui/services/forms.py [hunk 2]
```diff
@@ -1730,7 +1730,13 @@ def save(self):
             Samba4().set_administrator_password()
 
 
-class WebDAVForm(ModelForm):
+class WebDAVForm(MiddlewareModelForm, ModelForm):
+    middleware_attr_prefix = 'webdav_'
+    middleware_attr_schema = 'webdav'
```
[the remark this answers is not in the document]
Comment (themylogin) [reply gui/services/forms.py]: @william-gr why?

[the remark this answers is not in the document]
Comment (william-gr) [reply gui/services/forms.py]: @themylogin shouldn't it refer to the name of the Schema in do_update?

[the remark this answers is not in the document]
Comment (themylogin) [reply gui/services/forms.py]: @william-gr It is used as
```
            if form.middleware_attr_schema:
                schema = f'{form.middleware_attr_schema}_{form._middleware_action}'
```
So `create` forms are also validated successfully

[the remark this answers is not in the document]
Comment (william-gr) [reply gui/services/forms.py]: Ok, then nvm :)

**Pull request merged**